server.go

```diff
@@ -377,6 +380,49 @@ func main() {
 	}
 	fmt.Println("WorkspaceGateway provided")
 
+	// Provide OpenSearch Client
+	err = c.Provide(func() (*opensearch.Client, error) {
+		insecureSkipVerify := config.AppEnv() == constants.Development
+		cfg := opensearch.Config{
+			Addresses: []string{
+				config.OpenSearchURL(),
+			},
+			Username: config.OpenSearchUsername(),
+			Password: config.OpenSearchPassword(),
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: insecureSkipVerify,
+				},
+			},
+		}
+		openSearchClient, err := opensearch.NewClient(cfg)
+
+		fmt.Println("OpenSearch Client created....")
+		fmt.Println("OpenSearch Client: ", openSearchClient)
+		fmt.Println("Address: ", config.OpenSearchURL())
+		fmt.Println("Username: ", config.OpenSearchUsername())
+		fmt.Println("Password: ", config.OpenSearchPassword())
+
+		if err != nil {
+			return nil, fmt.Errorf("error creating the OpenSearch client: %w", err)
+		}
+		return openSearchClient, nil
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Provide OpenSearch Repository
+	err = c.Provide(func(client *opensearch.Client, logger *zap.Logger) repository.CodeBaseSearchRepository {
```

Comment:
Simplify provide

Reply:
This will be needed as which implementation of the repository has to be injected has to be decided

Reply:
"Will be", not needed now.